src/ponyfills/objects.js

```diff
@@ -156,9 +156,43 @@ export function prune(map, active) {
       }
     }
   })
+
+  reattributePredicatedValues(pruned, active);
   return pruned;
 }
 
+export function reattributePredicatedValues(pruned, active) {
```

Comment:
This function helps to produce a view of the genome with the matched predicated variant replacing the variant representing the several possible predicated variants.

 For example a genome retrieved from the allocations endpoint looks like this:
```json
{
	"web": {
		"k3iq091w0": {
			"dile18oq2": {
				"_predicated_variants_group_id": "1",
				"_predicated_values": [{
						"_predicate": {
							"combinator": "and",
							"rules": [{
								"field": "state",
								"operator": "equal",
								"value": "TX"
							}]
						},
						"_assignment_id": "a12345",
						"_value": {
							"id": "g4r6rg0w4",
							"type": "compound",
							"_metadata": {},
							"script": "console.log('red');",
							"styles": "#learn {background-color: red;}"
						}
					},
					{
						"_predicate": {
							"combinator": "and",
							"rules": [{
								"field": "state",
								"operator": "equal",
								"value": "CA"
							}]
						},
						"_assignment_id": "a56789",
						"_value": {
							"id": "rikee4btu",
							"type": "compound",
							"_metadata": {},
							"script": "console.log('blue');",
							"styles": "#learn {background-color: blue;}"
						}
					}
				],
				"_default_value": {
					"id": "p87zrhe2z",
					"type": "compound",
					"_metadata": {},
					"script": "console.log('green');",
					"styles": "#learn {background-color: green;}"
				}
			},
			"4p6q48f8h": {
				"type": "noop"
			}
		}
	}
}
```

After evaluating the context on the page and matching it with the predicated variants predicates, the Texas variant is deemed to be active, and so we generate the genome:
```json
{
	"web": {
		"k3iq091w0": {
			"dile18oq2": {
				"id": "g4r6rg0w4",
				"type": "compound",
				"_metadata": {},
				"script": "console.log('red');",
				"styles": "#learn {background-color: red;}"
			},
			"4p6q48f8h": {
				"type": "noop"
			}
		}
	}
}
```

Reply:
Its important to not that the function `evaluateAllocationPredicates` that is mentioned below runs before this function and populates the `active` set that this function relies upon.